Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptIDLookup.java

```
@@ -84,35 +84,38 @@
         }
         ConceptEntry entry = dictManager.getConceptEntry(wordnetId);
         Map<ConceptEntry, ConceptType> entryMap = new HashMap<ConceptEntry, ConceptType>();
-        List<String> xmlEntries = new ArrayList<String>();
 
-        XMLConceptMessage msg = messageFactory.createXMLConceptMessage();
+        IConceptMessage msg = messageFactory.getMessageFactory(acceptHeader).createConceptMessage();
 
         if (entry != null) {
 
             try {
-                // Check if the id used in a generic id. If so fetch the concept
-                // wrapper id for the generic wordnet id
-                if (conceptTypesService
-                        .getConceptTypeByConceptId(pathParts[lastIndex]) == ConceptTypes.GENERIC_WORDNET_CONCEPT) {
-                    entry = conceptManager.getConceptEntry(entry.getId());
-                }
-                addAlternativeIds(pathParts[lastIndex], entry);
+                // Check if the id used in a generic id. If so fetch the
+                // concept wrapper id for the generic wordnet id
+                entry = checkAndAddAlternativeIds(pathParts, lastIndex, entry);
             } catch (LuceneException e) {
                 logger.error("Lucene Exception", e);
-                return new ResponseEntity<String>(msg.getXML(xmlEntries), HttpStatus.INTERNAL_SERVER_ERROR);
+                return new ResponseEntity<String>(msg.getAllConceptMessage(entryMap), HttpStatus.INTERNAL_SERVER_ERROR);
```

This should return an error message.